src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -76,6 +78,59 @@ class WLSService(
         return movedItem
     }
 
+    override suspend fun pickItems(
+        hostName: HostName,
+        itemsPickedMap: Map<String, Double>
+    ) {
+        val itemIds =
+            itemsPickedMap.map {
+                ItemId(hostName, it.key)
+            }
+
+        if (!itemRepository.doesEveryItemExist(itemIds)) {
+            throw ItemNotFoundException("Some items do not exist in the database, and were unable to be picked")
+        }
+
+        itemIds.map { itemId ->
+            val item = getItem(itemId.hostName, itemId.hostId)!!
+            if (itemsPickedMap[item.hostId] == null) {
+                // TODO - Review this. This state should generally not happen with the preconditions. Does it need to be handled at all?
+                throw ItemNotFoundException("Item with ID ${itemId.hostId} for host ${item.hostName} was not found, despite existing!")
+            }
+            val movedItem =
+                itemRepository.moveItem(
+                    item.hostId,
+                    item.hostName,
+                    item.quantity?.minus((itemsPickedMap[item.hostId] ?: 0.0)) ?: 0.0,
+                    "Picked"
+                )
+            inventoryNotifier.itemChanged(movedItem)
+        }
+        // TODO - Should we log this once completed?
+    }
+
+    override suspend fun pickOrderItems(
+        hostName: HostName,
+        hostIds: List<String>,
+        orderId: String
+    ) {
+        // Make a new order line with picked items
+        val order = getOrder(hostName, orderId) ?: throw OrderNotFoundException("Order $orderId for host $hostName not found")
+
+        val orderLine =
+            order.orderLine.map { orderItem ->
+                hostIds.map { hostId ->
+                    if (orderItem.hostId == hostId) {
```

Comment:
Wouldn't something along these lines work too?

```kotlin
val newOrderLines = order.orderLine.map {
    if (hostIds.contains(it.hostId) {
        it.copy(status = Order.OrderItem.Status.PICKED)
    } else {
        it
    }
}
```
That way you could keep the status as a val not var and I think it's a bit clearer what's going on.